packages/builtin/asio_event.pony

```diff
@@ -7,6 +12,23 @@ primitive AsioEvent
   """
   Functions for asynchronous event notification.
   """
+
+  fun create_event(
+    owner: AsioEventNotify,
+    fd: U32,
+    flags: U32,
+    nsec: U64,
+    noisy: Bool)
+    : AsioEventID
+  =>
+    @pony_asio_event_create(owner, fd, flags, nsec, noisy)
+
+  fun unsubscribe(event: AsioEventID) =>
+    @pony_asio_event_unsubscribe(event)
+
+  fun destroy(event: AsioEventID) =>
+    @pony_asio_event_destroy(event)
+
```

Comment:
I've exposed these since they are used in several places. But I don't know if that's appropriate, maybe we don't want to expose this to regular users? (That is, keep it the way it was before).

Reply:
It may be a problem because everything that can be called from pure Pony without the caller using FFI is required to enforce memory safety through the constraints of its API.

My guess is that `pony_asio_event_create` is not safe to expose this way because you could provide a junk `fd`, which probably breaks memory safety. I also suspect that `pony_asio_event_destroy` is not safe because you could use it to double-free. Not sure about `pony_asio_event_unsubscribe` without looking deeper.

But given that at least those two should not be exposed, I recommend reverting this change and keeping these APIs as FFI-only, so that users must explicitly make a choice to take memory safety into their own hands if they use these APIs.

Reply:
I did this in lori. I think this is fine. You can already do this, just without a Pony wrapper. Any issues that exist already exist. This won't add any issues.

I think if we have concerns about safety issues, we need to fix those.

See in Lori:

https://github.com/seantallen-org/lori/blob/main/lori/pony_asio.pony

Reply:
@jemc I don't see how `pony_asio_event_destroy` can be used to double free unless you have concurrent access to the same Event at the same time.

From the method:

```c
  if((ev == NULL) || (ev->magic != ev) || (ev->flags != ASIO_DISPOSABLE))
  {
    pony_assert(0);
    return;
  }
```

So once it has been freed, its NULL and will return.

Reply:
`pony_asio_event_create` doesn't from what I can see have any issues with safety that every POSIX file API that uses fd's would have.

Reply:
I don't see any problems with `pony_asio_event_unsubscribe` that would cause me to worry.

I think we want to expose all of these as part of "making it friendlier to work with the ASIO subsystem", however, that isn't the goal of this PR, so I'd be quite fine with not including these changes in this PR.

Reply:
Okay, I'll revert this change. We can talk about exposing this at some other time.

Reply:
Done